### PR TITLE
feat: Re-export "PreviewGallery" component

### PR DESCRIPTION
**Pull request by Yakutoc**
Необходимо: 

1. Перенести компонент из **plasma-b2c** в **plasma-hope** и сделать **re-export** в 

   -  **plasma-b2c**
   -  **plasma-web**

2. Проверить дизайн в **plasma-web**;

:baby_chick: Download canary assets:

[colors--canary.376.4312034677.xml](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/colors--canary.376.4312034677.xml)  
[color_metro_ios-swift--canary.376.4312034677.swift](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_metro_ios-swift--canary.376.4312034677.swift)  
[color_metro_kotlin--canary.376.4312034677.kt](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_metro_kotlin--canary.376.4312034677.kt)  
[color_metro_react-native--canary.376.4312034677.ts](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_metro_react-native--canary.376.4312034677.ts)  
[color_sberHealth_ios-swift--canary.376.4312034677.swift](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_sberHealth_ios-swift--canary.376.4312034677.swift)  
[color_sberHealth_kotlin--canary.376.4312034677.kt](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_sberHealth_kotlin--canary.376.4312034677.kt)  
[color_sberHealth_react-native--canary.376.4312034677.ts](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_sberHealth_react-native--canary.376.4312034677.ts)  
[color_sbermarket_ios-swift--canary.376.4312034677.swift](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_sbermarket_ios-swift--canary.376.4312034677.swift)  
[color_sbermarket_kotlin--canary.376.4312034677.kt](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_sbermarket_kotlin--canary.376.4312034677.kt)  
[color_sbermarket_react-native--canary.376.4312034677.ts](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_sbermarket_react-native--canary.376.4312034677.ts)  
[color_sberprime_ios-swift--canary.376.4312034677.swift](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_sberprime_ios-swift--canary.376.4312034677.swift)  
[color_sberprime_kotlin--canary.376.4312034677.kt](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_sberprime_kotlin--canary.376.4312034677.kt)  
[color_sberprime_react-native--canary.376.4312034677.ts](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_sberprime_react-native--canary.376.4312034677.ts)  
[color_selgros_ios-swift--canary.376.4312034677.swift](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_selgros_ios-swift--canary.376.4312034677.swift)  
[color_selgros_kotlin--canary.376.4312034677.kt](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_selgros_kotlin--canary.376.4312034677.kt)  
[color_selgros_react-native--canary.376.4312034677.ts](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_selgros_react-native--canary.376.4312034677.ts)  
[color_smbusiness_ios-swift--canary.376.4312034677.swift](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_smbusiness_ios-swift--canary.376.4312034677.swift)  
[color_smbusiness_kotlin--canary.376.4312034677.kt](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_smbusiness_kotlin--canary.376.4312034677.kt)  
[color_smbusiness_react-native--canary.376.4312034677.ts](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_smbusiness_react-native--canary.376.4312034677.ts)  
[PlasmaTokensColor--canary.376.4312034677.swift](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/PlasmaTokensColor--canary.376.4312034677.swift)  
[shadow_sbermarket_react-native--canary.376.4312034677.ts](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/shadow_sbermarket_react-native--canary.376.4312034677.ts)  
[typo_mage_ios-swift--canary.376.4312034677.swift](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_mage_ios-swift--canary.376.4312034677.swift)  
[typo_mage_kotlin--canary.376.4312034677.kt](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_mage_kotlin--canary.376.4312034677.kt)  
[typo_mage_react-native--canary.376.4312034677.ts](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_mage_react-native--canary.376.4312034677.ts)  
[typo_plasma_ios-swift--canary.376.4312034677.swift](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_plasma_ios-swift--canary.376.4312034677.swift)  
[typo_plasma_kotlin--canary.376.4312034677.kt](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_plasma_kotlin--canary.376.4312034677.kt)  
[typo_plasma_react-native--canary.376.4312034677.ts](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_plasma_react-native--canary.376.4312034677.ts)  
[typo_ruler_ios-swift--canary.376.4312034677.swift](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_ruler_ios-swift--canary.376.4312034677.swift)  
[typo_ruler_kotlin--canary.376.4312034677.kt](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_ruler_kotlin--canary.376.4312034677.kt)  
[typo_ruler_react-native--canary.376.4312034677.ts](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_ruler_react-native--canary.376.4312034677.ts)  
[typo_sage_ios-swift--canary.376.4312034677.swift](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_sage_ios-swift--canary.376.4312034677.swift)  
[typo_sage_kotlin--canary.376.4312034677.kt](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_sage_kotlin--canary.376.4312034677.kt)  
[typo_sage_react-native--canary.376.4312034677.ts](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_sage_react-native--canary.376.4312034677.ts)  
[typo_sbermarket_ios-swift--canary.376.4312034677.swift](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_sbermarket_ios-swift--canary.376.4312034677.swift)  
[typo_sbermarket_kotlin--canary.376.4312034677.kt](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_sbermarket_kotlin--canary.376.4312034677.kt)  
[typo_sbermarket_react-native--canary.376.4312034677.ts](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_sbermarket_react-native--canary.376.4312034677.ts)  
[typo_soulmate_ios-swift--canary.376.4312034677.swift](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_soulmate_ios-swift--canary.376.4312034677.swift)  
[typo_soulmate_kotlin--canary.376.4312034677.kt](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_soulmate_kotlin--canary.376.4312034677.kt)  
[typo_soulmate_react-native--canary.376.4312034677.ts](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_soulmate_react-native--canary.376.4312034677.ts)

<details>
  <summary>📦 Published PR as canary version: <code>Canary Versions</code></summary>
  <br />

  :sparkles: Test out this PR locally via:
  
  ```bash
  npm install @salutejs/plasma-b2c@1.148.0-canary.376.4312034677.0
  npm install @salutejs/plasma-hope@0.7.0-canary.376.4312034677.0
  npm install @salutejs/plasma-web@1.173.0-canary.376.4312034677.0
  # or 
  yarn add @salutejs/plasma-b2c@1.148.0-canary.376.4312034677.0
  yarn add @salutejs/plasma-hope@0.7.0-canary.376.4312034677.0
  yarn add @salutejs/plasma-web@1.173.0-canary.376.4312034677.0
  ```
</details>

### CODE REVIEW

**Comment by Salute-Eva** [comment]
Theme Builder app deployed!
  
  http://plasma.sberdevices.ru/pr/plasma-theme-builder-pr-376/

**Comment by Salute-Eva** [comment]
Demo canvas app deployed!
  
  http://plasma.sberdevices.ru/pr/demo-canvas-app-pr-376/

**Comment by Salute-Eva** [comment]
**Component performance testing**

Result: **🟢 Success**

Check out comparison report in job artifacts!

File: packages/plasma-b2c/src/components/PreviewGallery/PreviewGallery.stories.tsx [hunk 1]
```diff
@@ -1,16 +1,19 @@
-import React, { useState, useCallback } from 'react';
+import React, { useCallback, useState } from 'react';
 import styled from 'styled-components';
-import { Story, Meta } from '@storybook/react';
+import { Meta, Story } from '@storybook/react';
 import { InSpacingDecorator } from '@salutejs/plasma-sb-utils';
-import { IconTrashFilled, IconTrash } from '@salutejs/plasma-icons';
+import { IconTrash, IconTrashFilled } from '@salutejs/plasma-icons';
```
Comment (Yeti-or) [inline packages/plasma-b2c/src/components/PreviewGallery/PreviewGallery.stories.tsx:5]: это плагин какой-то переставляет? или ты ручками?

File: packages/plasma-b2c/src/components/PreviewGallery/PreviewGallery.stories.tsx [hunk 2]
```diff
@@ -63,7 +66,7 @@ interface StoryProps extends PreviewGalleryProps {}
 
 const images: Array<PreviewGalleryItemProps> = [
     {
-        id: 1 * Math.random(),
+        id: Math.random(),
```
Comment (Yeti-or) [inline packages/plasma-b2c/src/components/PreviewGallery/PreviewGallery.stories.tsx:69]: `Math.floor(100 * Math.random())` лучше уж так

File: packages/plasma-hope/package.json [hunk 3]
```diff
@@ -78,6 +62,7 @@
         "react": "16.13.1",
         "react-dom": "16.13.1",
         "react-scripts": "3.4.4",
+        "react-sortable-hoc": "2.0.0",
```
Comment (Yeti-or) [inline packages/plasma-hope/package.json:65]: а из b2c это не надо теперь удалить?

**Review by Yeti-or — APPROVED** [review]
я кстати понял что их нужно будет от друг друга ребейзить и вливать по одному каждый раз обновляя diff и api.md

File: packages/plasma-b2c/src/components/PreviewGallery/PreviewGallery.stories.tsx (hunk 1 above)
Comment (Yakutoc) [reply packages/plasma-b2c/src/components/PreviewGallery/PreviewGallery.stories.tsx:5]: Это скорее всего WebStorm делает сортировку.

**Comment by Yakutoc** [comment]
> я кстати понял что их нужно будет от друг друга ребейзить и вливать по одному каждый раз обновляя diff и api.md

Да, именно так.

File: packages/plasma-hope/package.json (hunk 3 above)
Comment (Yakutoc) [reply packages/plasma-hope/package.json:65]: Да, будет отделельнная задача на выпил лишних зависимостей из b2

File: packages/plasma-b2c/src/components/PreviewGallery/PreviewGallery.stories.tsx (hunk 2 above)
Comment (Yakutoc) [reply packages/plasma-b2c/src/components/PreviewGallery/PreviewGallery.stories.tsx:69]: Когда буду делать rebase - поправлю этот момент.

**Review by gibushnev — APPROVED** [review]

**Review by neretin-trike — APPROVED** [review]

**Comment by Salute-Eva** [comment]
Theme Builder app deployed!
  
  http://plasma.sberdevices.ru/pr/plasma-theme-builder-pr-376/

**Comment by Salute-Eva** [comment]
Demo canvas app deployed!
  
  http://plasma.sberdevices.ru/pr/demo-canvas-app-pr-376/

**Comment by Salute-Eva** [comment]
Documentation preview deployed!
  
  website: http://plasma.sberdevices.ru/pr/pr-376/
  ui storybook: http://plasma.sberdevices.ru/pr/pr-376/ui-storybook/
  web storybook: http://plasma.sberdevices.ru/pr/pr-376/web-storybook/
  b2c storybook: http://plasma.sberdevices.ru/pr/pr-376/b2c-storybook/
  temple storybook: http://plasma.sberdevices.ru/pr/pr-376/temple-storybook/

**Pull request merged**